### PR TITLE
Updated with the latest SDL3 changes

### PR DESCRIPTION
**Pull request by slouken**
SDL_SetWindowShape() has returned in SDL 3.0

Use the property definitions instead of hard-coded strings

### CODE REVIEW

File: src/sdl2_compat.c [hunk 1]
```diff
@@ -3527,153 +3527,89 @@ GestureProcessEvent(const SDL_Event *event3)
...
         return SDL_TRUE;
     }
     return SDL_FALSE;
 }
 
+#define PROP_WINDOW_SHAPE_MODE_POINTER "sdl2-compat.window.shape_mode"
+
+static void CleanupFreeableProperty(void *userdata, void *value)
+{
+    SDL_free(value);
```
Comment (sezero) [inline src/sdl2_compat.c:3603]: SDL3_free

Comment (slouken) [reply src/sdl2_compat.c]: Fixed

**Comment by slouken** [comment]
Good to go?

File: src/sdl2_compat.c [hunk 2]
```diff
@@ -3527,153 +3527,89 @@ GestureProcessEvent(const SDL_Event *event3)
...
-    if (window == g_shaped_window) {
+    if (SDL3_GetWindowFlags((SDL_Window *)window) & SDL_WINDOW_TRANSPARENT) {
         return SDL_TRUE;
     }
     return SDL_FALSE;
 }
 
+#define PROP_WINDOW_SHAPE_MODE_POINTER "sdl2-compat.window.shape_mode"
+
+static void CleanupFreeableProperty(void *userdata, void *value)
```
Comment (sezero) [inline src/sdl2_compat.c:3601]: This one must be `SDLCALL` -- i.e.:
`static void SDLCALL CleanupFreeableProperty(void *userdata, void *value)`

Comment (slouken) [reply src/sdl2_compat.c]: Fixed

**Comment by sezero** [comment]
> Good to go?

Yes, as far as I can see

**Comment by slouken** [comment]
Thanks!

**Pull request merged**